applications/plugins/SofaPython/Binding_Data.cpp

```diff
@@ -670,6 +674,26 @@ bool SetDataValuePython(BaseData* data, PyObject* args)
 }
 
 
+SP_CLASS_ATTR_GET(Data, fullPath)(PyObject *self, void*)
+{
+    BaseData* data = ((PyPtr<BaseData>*)self)->object; // TODO: check dynamic cast
+
+    if (sofa::core::objectmodel::DataFileName* dataFilename = dynamic_cast<sofa::core::objectmodel::DataFileName*>(data))
+    {
```

Comment:
The original commit predates the concept of the `PythonFactory`. So it could indeed be refactored now, by someone who knows how to use this Factory.